// @aws-sdk/client-quicksight/dist/es/commands/CreateTemplateAliasCommand.js
import { __extends } from "tslib";
import { CreateTemplateAliasRequest, CreateTemplateAliasResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateTemplateAliasCommand, serializeAws_restJson1CreateTemplateAliasCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Creates a template alias for a template.</p>
 */
var CreateTemplateAliasCommand = /** @class */ (function (_super) {
    __extends(CreateTemplateAliasCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CreateTemplateAliasCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    CreateTemplateAliasCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "QuickSightClient";
        var commandName = "CreateTemplateAliasCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateTemplateAliasRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateTemplateAliasResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateTemplateAliasCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateTemplateAliasCommand(input, context);
    };
    CreateTemplateAliasCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateTemplateAliasCommand(output, context);
    };
    return CreateTemplateAliasCommand;
}($Command));
export { CreateTemplateAliasCommand };
