// @aws-sdk/client-quicksight/dist/es/commands/CreateDataSetCommand.js
import { __extends } from "tslib";
import { CreateDataSetRequest, CreateDataSetResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateDataSetCommand, serializeAws_restJson1CreateDataSetCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Creates a dataset.</p>
 */
var CreateDataSetCommand = /** @class */ (function (_super) {
    __extends(CreateDataSetCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CreateDataSetCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    CreateDataSetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "QuickSightClient";
        var commandName = "CreateDataSetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDataSetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateDataSetResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDataSetCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateDataSetCommand(input, context);
    };
    CreateDataSetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateDataSetCommand(output, context);
    };
    return CreateDataSetCommand;
}($Command));
export { CreateDataSetCommand };
