// @aws-sdk/client-quicksight/dist/es/commands/CreateIAMPolicyAssignmentCommand.js
import { __extends } from "tslib";
import { CreateIAMPolicyAssignmentRequest, CreateIAMPolicyAssignmentResponse } from "../models/models_0";
import { deserializeAws_restJson1CreateIAMPolicyAssignmentCommand, serializeAws_restJson1CreateIAMPolicyAssignmentCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Creates an assignment with one specified IAM policy, identified by its Amazon Resource Name
 * 			(ARN). This policy assignment is attached to the specified groups or users of Amazon QuickSight.
 * 			Assignment names are unique per AWS account. To avoid overwriting rules in other namespaces,
 * 			use assignment names that are unique.</p>
 */
var CreateIAMPolicyAssignmentCommand = /** @class */ (function (_super) {
    __extends(CreateIAMPolicyAssignmentCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CreateIAMPolicyAssignmentCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    CreateIAMPolicyAssignmentCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "QuickSightClient";
        var commandName = "CreateIAMPolicyAssignmentCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateIAMPolicyAssignmentRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateIAMPolicyAssignmentResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateIAMPolicyAssignmentCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateIAMPolicyAssignmentCommand(input, context);
    };
    CreateIAMPolicyAssignmentCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateIAMPolicyAssignmentCommand(output, context);
    };
    return CreateIAMPolicyAssignmentCommand;
}($Command));
export { CreateIAMPolicyAssignmentCommand };
