// @aws-sdk/client-quicksight/dist/es/commands/DescribeDataSetCommand.js
import { __extends } from "tslib";
import { DescribeDataSetRequest, DescribeDataSetResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeDataSetCommand, serializeAws_restJson1DescribeDataSetCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Describes a dataset. </p>
 */
var DescribeDataSetCommand = /** @class */ (function (_super) {
    __extends(DescribeDataSetCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function DescribeDataSetCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    DescribeDataSetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "QuickSightClient";
        var commandName = "DescribeDataSetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDataSetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeDataSetResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDataSetCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeDataSetCommand(input, context);
    };
    DescribeDataSetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeDataSetCommand(output, context);
    };
    return DescribeDataSetCommand;
}($Command));
export { DescribeDataSetCommand };
