// @aws-sdk/client-quicksight/dist/es/commands/ListNamespacesCommand.js
import { __extends } from "tslib";
import { ListNamespacesRequest, ListNamespacesResponse } from "../models/models_1";
import { deserializeAws_restJson1ListNamespacesCommand, serializeAws_restJson1ListNamespacesCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Lists the namespaces for the specified AWS account.</p>
 */
var ListNamespacesCommand = /** @class */ (function (_super) {
    __extends(ListNamespacesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListNamespacesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListNamespacesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "QuickSightClient";
        var commandName = "ListNamespacesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListNamespacesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListNamespacesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListNamespacesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListNamespacesCommand(input, context);
    };
    ListNamespacesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListNamespacesCommand(output, context);
    };
    return ListNamespacesCommand;
}($Command));
export { ListNamespacesCommand };
