// @aws-sdk/client-quicksight/dist/es/commands/ListGroupMembershipsCommand.js
import { __extends } from "tslib";
import { ListGroupMembershipsRequest, ListGroupMembershipsResponse } from "../models/models_0";
import { deserializeAws_restJson1ListGroupMembershipsCommand, serializeAws_restJson1ListGroupMembershipsCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
/**
 * <p>Lists member users in a group.</p>
 */
var ListGroupMembershipsCommand = /** @class */ (function (_super) {
    __extends(ListGroupMembershipsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListGroupMembershipsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    /**
     * @internal
     */
    ListGroupMembershipsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "QuickSightClient";
        var commandName = "ListGroupMembershipsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListGroupMembershipsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListGroupMembershipsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListGroupMembershipsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListGroupMembershipsCommand(input, context);
    };
    ListGroupMembershipsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListGroupMembershipsCommand(output, context);
    };
    return ListGroupMembershipsCommand;
}($Command));
export { ListGroupMembershipsCommand };
