// @aws-sdk/client-quicksight/dist/cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDashboardPublishedVersionResponse = exports.UpdateDashboardPublishedVersionRequest = exports.UpdateDashboardPermissionsResponse = exports.UpdateDashboardPermissionsRequest = exports.UpdateDashboardResponse = exports.UpdateDashboardRequest = exports.UpdateAnalysisPermissionsResponse = exports.UpdateAnalysisPermissionsRequest = exports.UpdateAnalysisResponse = exports.UpdateAnalysisRequest = exports.UpdateAccountSettingsResponse = exports.UpdateAccountSettingsRequest = exports.UpdateAccountCustomizationResponse = exports.UpdateAccountCustomizationRequest = exports.UntagResourceResponse = exports.UntagResourceRequest = exports.TagResourceResponse = exports.TagResourceRequest = exports.SearchDashboardsResponse = exports.SearchDashboardsRequest = exports.SearchAnalysesResponse = exports.SearchAnalysesRequest = exports.RestoreAnalysisResponse = exports.RestoreAnalysisRequest = exports.RegisterUserResponse = exports.RegisterUserRequest = exports.ListUsersResponse = exports.ListUsersRequest = exports.ListUserGroupsResponse = exports.ListUserGroupsRequest = exports.ListThemeVersionsResponse = exports.ThemeVersionSummary = exports.ListThemeVersionsRequest = exports.ListThemesResponse = exports.ThemeSummary = exports.ListThemesRequest = exports.ListThemeAliasesResponse = exports.ListThemeAliasesRequest = exports.ListTemplateVersionsResponse = exports.TemplateVersionSummary = exports.ListTemplateVersionsRequest = exports.ListTemplatesResponse = exports.TemplateSummary = exports.ListTemplatesRequest = exports.ListTemplateAliasesResponse = exports.ListTemplateAliasesRequest = exports.ListTagsForResourceResponse = exports.ListTagsForResourceRequest = exports.ListNamespacesResponse = exports.ListNamespacesRequest = void 0;
exports.UpdateUserResponse = exports.UpdateUserRequest = exports.UpdateThemePermissionsResponse = exports.UpdateThemePermissionsRequest = exports.UpdateThemeAliasResponse = exports.UpdateThemeAliasRequest = exports.UpdateThemeResponse = exports.UpdateThemeRequest = exports.UpdateTemplatePermissionsResponse = exports.UpdateTemplatePermissionsRequest = exports.UpdateTemplateAliasResponse = exports.UpdateTemplateAliasRequest = exports.UpdateTemplateResponse = exports.UpdateTemplateRequest = exports.UpdateIAMPolicyAssignmentResponse = exports.UpdateIAMPolicyAssignmentRequest = exports.UpdateGroupResponse = exports.UpdateGroupRequest = exports.UpdateDataSourcePermissionsResponse = exports.UpdateDataSourcePermissionsRequest = exports.UpdateDataSourceResponse = exports.UpdateDataSourceRequest = exports.UpdateDataSetPermissionsResponse = exports.UpdateDataSetPermissionsRequest = exports.UpdateDataSetResponse = exports.UpdateDataSetRequest = void 0;
const models_0_1 = require("./models_0");
const smithy_client_1 = require("@aws-sdk/smithy-client");
var ListNamespacesRequest;
(function (ListNamespacesRequest) {
    ListNamespacesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListNamespacesRequest = exports.ListNamespacesRequest || (exports.ListNamespacesRequest = {}));
var ListNamespacesResponse;
(function (ListNamespacesResponse) {
    ListNamespacesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListNamespacesResponse = exports.ListNamespacesResponse || (exports.ListNamespacesResponse = {}));
var ListTagsForResourceRequest;
(function (ListTagsForResourceRequest) {
    ListTagsForResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceRequest = exports.ListTagsForResourceRequest || (exports.ListTagsForResourceRequest = {}));
var ListTagsForResourceResponse;
(function (ListTagsForResourceResponse) {
    ListTagsForResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTagsForResourceResponse = exports.ListTagsForResourceResponse || (exports.ListTagsForResourceResponse = {}));
var ListTemplateAliasesRequest;
(function (ListTemplateAliasesRequest) {
    ListTemplateAliasesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTemplateAliasesRequest = exports.ListTemplateAliasesRequest || (exports.ListTemplateAliasesRequest = {}));
var ListTemplateAliasesResponse;
(function (ListTemplateAliasesResponse) {
    ListTemplateAliasesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTemplateAliasesResponse = exports.ListTemplateAliasesResponse || (exports.ListTemplateAliasesResponse = {}));
var ListTemplatesRequest;
(function (ListTemplatesRequest) {
    ListTemplatesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTemplatesRequest = exports.ListTemplatesRequest || (exports.ListTemplatesRequest = {}));
var TemplateSummary;
(function (TemplateSummary) {
    TemplateSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TemplateSummary = exports.TemplateSummary || (exports.TemplateSummary = {}));
var ListTemplatesResponse;
(function (ListTemplatesResponse) {
    ListTemplatesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTemplatesResponse = exports.ListTemplatesResponse || (exports.ListTemplatesResponse = {}));
var ListTemplateVersionsRequest;
(function (ListTemplateVersionsRequest) {
    ListTemplateVersionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTemplateVersionsRequest = exports.ListTemplateVersionsRequest || (exports.ListTemplateVersionsRequest = {}));
var TemplateVersionSummary;
(function (TemplateVersionSummary) {
    TemplateVersionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TemplateVersionSummary = exports.TemplateVersionSummary || (exports.TemplateVersionSummary = {}));
var ListTemplateVersionsResponse;
(function (ListTemplateVersionsResponse) {
    ListTemplateVersionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTemplateVersionsResponse = exports.ListTemplateVersionsResponse || (exports.ListTemplateVersionsResponse = {}));
var ListThemeAliasesRequest;
(function (ListThemeAliasesRequest) {
    ListThemeAliasesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThemeAliasesRequest = exports.ListThemeAliasesRequest || (exports.ListThemeAliasesRequest = {}));
var ListThemeAliasesResponse;
(function (ListThemeAliasesResponse) {
    ListThemeAliasesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThemeAliasesResponse = exports.ListThemeAliasesResponse || (exports.ListThemeAliasesResponse = {}));
var ListThemesRequest;
(function (ListThemesRequest) {
    ListThemesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThemesRequest = exports.ListThemesRequest || (exports.ListThemesRequest = {}));
var ThemeSummary;
(function (ThemeSummary) {
    ThemeSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ThemeSummary = exports.ThemeSummary || (exports.ThemeSummary = {}));
var ListThemesResponse;
(function (ListThemesResponse) {
    ListThemesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThemesResponse = exports.ListThemesResponse || (exports.ListThemesResponse = {}));
var ListThemeVersionsRequest;
(function (ListThemeVersionsRequest) {
    ListThemeVersionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThemeVersionsRequest = exports.ListThemeVersionsRequest || (exports.ListThemeVersionsRequest = {}));
var ThemeVersionSummary;
(function (ThemeVersionSummary) {
    ThemeVersionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ThemeVersionSummary = exports.ThemeVersionSummary || (exports.ThemeVersionSummary = {}));
var ListThemeVersionsResponse;
(function (ListThemeVersionsResponse) {
    ListThemeVersionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListThemeVersionsResponse = exports.ListThemeVersionsResponse || (exports.ListThemeVersionsResponse = {}));
var ListUserGroupsRequest;
(function (ListUserGroupsRequest) {
    ListUserGroupsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUserGroupsRequest = exports.ListUserGroupsRequest || (exports.ListUserGroupsRequest = {}));
var ListUserGroupsResponse;
(function (ListUserGroupsResponse) {
    ListUserGroupsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUserGroupsResponse = exports.ListUserGroupsResponse || (exports.ListUserGroupsResponse = {}));
var ListUsersRequest;
(function (ListUsersRequest) {
    ListUsersRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUsersRequest = exports.ListUsersRequest || (exports.ListUsersRequest = {}));
var ListUsersResponse;
(function (ListUsersResponse) {
    ListUsersResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListUsersResponse = exports.ListUsersResponse || (exports.ListUsersResponse = {}));
var RegisterUserRequest;
(function (RegisterUserRequest) {
    RegisterUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RegisterUserRequest = exports.RegisterUserRequest || (exports.RegisterUserRequest = {}));
var RegisterUserResponse;
(function (RegisterUserResponse) {
    RegisterUserResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RegisterUserResponse = exports.RegisterUserResponse || (exports.RegisterUserResponse = {}));
var RestoreAnalysisRequest;
(function (RestoreAnalysisRequest) {
    RestoreAnalysisRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreAnalysisRequest = exports.RestoreAnalysisRequest || (exports.RestoreAnalysisRequest = {}));
var RestoreAnalysisResponse;
(function (RestoreAnalysisResponse) {
    RestoreAnalysisResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RestoreAnalysisResponse = exports.RestoreAnalysisResponse || (exports.RestoreAnalysisResponse = {}));
var SearchAnalysesRequest;
(function (SearchAnalysesRequest) {
    SearchAnalysesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SearchAnalysesRequest = exports.SearchAnalysesRequest || (exports.SearchAnalysesRequest = {}));
var SearchAnalysesResponse;
(function (SearchAnalysesResponse) {
    SearchAnalysesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SearchAnalysesResponse = exports.SearchAnalysesResponse || (exports.SearchAnalysesResponse = {}));
var SearchDashboardsRequest;
(function (SearchDashboardsRequest) {
    SearchDashboardsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SearchDashboardsRequest = exports.SearchDashboardsRequest || (exports.SearchDashboardsRequest = {}));
var SearchDashboardsResponse;
(function (SearchDashboardsResponse) {
    SearchDashboardsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SearchDashboardsResponse = exports.SearchDashboardsResponse || (exports.SearchDashboardsResponse = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
var TagResourceResponse;
(function (TagResourceResponse) {
    TagResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceResponse = exports.TagResourceResponse || (exports.TagResourceResponse = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UntagResourceResponse;
(function (UntagResourceResponse) {
    UntagResourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceResponse = exports.UntagResourceResponse || (exports.UntagResourceResponse = {}));
var UpdateAccountCustomizationRequest;
(function (UpdateAccountCustomizationRequest) {
    UpdateAccountCustomizationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAccountCustomizationRequest = exports.UpdateAccountCustomizationRequest || (exports.UpdateAccountCustomizationRequest = {}));
var UpdateAccountCustomizationResponse;
(function (UpdateAccountCustomizationResponse) {
    UpdateAccountCustomizationResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAccountCustomizationResponse = exports.UpdateAccountCustomizationResponse || (exports.UpdateAccountCustomizationResponse = {}));
var UpdateAccountSettingsRequest;
(function (UpdateAccountSettingsRequest) {
    UpdateAccountSettingsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAccountSettingsRequest = exports.UpdateAccountSettingsRequest || (exports.UpdateAccountSettingsRequest = {}));
var UpdateAccountSettingsResponse;
(function (UpdateAccountSettingsResponse) {
    UpdateAccountSettingsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAccountSettingsResponse = exports.UpdateAccountSettingsResponse || (exports.UpdateAccountSettingsResponse = {}));
var UpdateAnalysisRequest;
(function (UpdateAnalysisRequest) {
    UpdateAnalysisRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAnalysisRequest = exports.UpdateAnalysisRequest || (exports.UpdateAnalysisRequest = {}));
var UpdateAnalysisResponse;
(function (UpdateAnalysisResponse) {
    UpdateAnalysisResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAnalysisResponse = exports.UpdateAnalysisResponse || (exports.UpdateAnalysisResponse = {}));
var UpdateAnalysisPermissionsRequest;
(function (UpdateAnalysisPermissionsRequest) {
    UpdateAnalysisPermissionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAnalysisPermissionsRequest = exports.UpdateAnalysisPermissionsRequest || (exports.UpdateAnalysisPermissionsRequest = {}));
var UpdateAnalysisPermissionsResponse;
(function (UpdateAnalysisPermissionsResponse) {
    UpdateAnalysisPermissionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAnalysisPermissionsResponse = exports.UpdateAnalysisPermissionsResponse || (exports.UpdateAnalysisPermissionsResponse = {}));
var UpdateDashboardRequest;
(function (UpdateDashboardRequest) {
    UpdateDashboardRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDashboardRequest = exports.UpdateDashboardRequest || (exports.UpdateDashboardRequest = {}));
var UpdateDashboardResponse;
(function (UpdateDashboardResponse) {
    UpdateDashboardResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDashboardResponse = exports.UpdateDashboardResponse || (exports.UpdateDashboardResponse = {}));
var UpdateDashboardPermissionsRequest;
(function (UpdateDashboardPermissionsRequest) {
    UpdateDashboardPermissionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDashboardPermissionsRequest = exports.UpdateDashboardPermissionsRequest || (exports.UpdateDashboardPermissionsRequest = {}));
var UpdateDashboardPermissionsResponse;
(function (UpdateDashboardPermissionsResponse) {
    UpdateDashboardPermissionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDashboardPermissionsResponse = exports.UpdateDashboardPermissionsResponse || (exports.UpdateDashboardPermissionsResponse = {}));
var UpdateDashboardPublishedVersionRequest;
(function (UpdateDashboardPublishedVersionRequest) {
    UpdateDashboardPublishedVersionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDashboardPublishedVersionRequest = exports.UpdateDashboardPublishedVersionRequest || (exports.UpdateDashboardPublishedVersionRequest = {}));
var UpdateDashboardPublishedVersionResponse;
(function (UpdateDashboardPublishedVersionResponse) {
    UpdateDashboardPublishedVersionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDashboardPublishedVersionResponse = exports.UpdateDashboardPublishedVersionResponse || (exports.UpdateDashboardPublishedVersionResponse = {}));
var UpdateDataSetRequest;
(function (UpdateDataSetRequest) {
    UpdateDataSetRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.PhysicalTableMap && {
            PhysicalTableMap: Object.entries(obj.PhysicalTableMap).reduce((acc, [key, value]) => ({
                ...acc,
                [key]: models_0_1.PhysicalTable.filterSensitiveLog(value),
            }), {}),
        }),
        ...(obj.LogicalTableMap && {
            LogicalTableMap: Object.entries(obj.LogicalTableMap).reduce((acc, [key, value]) => ({
                ...acc,
                [key]: models_0_1.LogicalTable.filterSensitiveLog(value),
            }), {}),
        }),
    });
})(UpdateDataSetRequest = exports.UpdateDataSetRequest || (exports.UpdateDataSetRequest = {}));
var UpdateDataSetResponse;
(function (UpdateDataSetResponse) {
    UpdateDataSetResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDataSetResponse = exports.UpdateDataSetResponse || (exports.UpdateDataSetResponse = {}));
var UpdateDataSetPermissionsRequest;
(function (UpdateDataSetPermissionsRequest) {
    UpdateDataSetPermissionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDataSetPermissionsRequest = exports.UpdateDataSetPermissionsRequest || (exports.UpdateDataSetPermissionsRequest = {}));
var UpdateDataSetPermissionsResponse;
(function (UpdateDataSetPermissionsResponse) {
    UpdateDataSetPermissionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDataSetPermissionsResponse = exports.UpdateDataSetPermissionsResponse || (exports.UpdateDataSetPermissionsResponse = {}));
var UpdateDataSourceRequest;
(function (UpdateDataSourceRequest) {
    UpdateDataSourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.DataSourceParameters && {
            DataSourceParameters: models_0_1.DataSourceParameters.filterSensitiveLog(obj.DataSourceParameters),
        }),
        ...(obj.Credentials && { Credentials: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateDataSourceRequest = exports.UpdateDataSourceRequest || (exports.UpdateDataSourceRequest = {}));
var UpdateDataSourceResponse;
(function (UpdateDataSourceResponse) {
    UpdateDataSourceResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDataSourceResponse = exports.UpdateDataSourceResponse || (exports.UpdateDataSourceResponse = {}));
var UpdateDataSourcePermissionsRequest;
(function (UpdateDataSourcePermissionsRequest) {
    UpdateDataSourcePermissionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDataSourcePermissionsRequest = exports.UpdateDataSourcePermissionsRequest || (exports.UpdateDataSourcePermissionsRequest = {}));
var UpdateDataSourcePermissionsResponse;
(function (UpdateDataSourcePermissionsResponse) {
    UpdateDataSourcePermissionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateDataSourcePermissionsResponse = exports.UpdateDataSourcePermissionsResponse || (exports.UpdateDataSourcePermissionsResponse = {}));
var UpdateGroupRequest;
(function (UpdateGroupRequest) {
    UpdateGroupRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateGroupRequest = exports.UpdateGroupRequest || (exports.UpdateGroupRequest = {}));
var UpdateGroupResponse;
(function (UpdateGroupResponse) {
    UpdateGroupResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateGroupResponse = exports.UpdateGroupResponse || (exports.UpdateGroupResponse = {}));
var UpdateIAMPolicyAssignmentRequest;
(function (UpdateIAMPolicyAssignmentRequest) {
    UpdateIAMPolicyAssignmentRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateIAMPolicyAssignmentRequest = exports.UpdateIAMPolicyAssignmentRequest || (exports.UpdateIAMPolicyAssignmentRequest = {}));
var UpdateIAMPolicyAssignmentResponse;
(function (UpdateIAMPolicyAssignmentResponse) {
    UpdateIAMPolicyAssignmentResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateIAMPolicyAssignmentResponse = exports.UpdateIAMPolicyAssignmentResponse || (exports.UpdateIAMPolicyAssignmentResponse = {}));
var UpdateTemplateRequest;
(function (UpdateTemplateRequest) {
    UpdateTemplateRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTemplateRequest = exports.UpdateTemplateRequest || (exports.UpdateTemplateRequest = {}));
var UpdateTemplateResponse;
(function (UpdateTemplateResponse) {
    UpdateTemplateResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTemplateResponse = exports.UpdateTemplateResponse || (exports.UpdateTemplateResponse = {}));
var UpdateTemplateAliasRequest;
(function (UpdateTemplateAliasRequest) {
    UpdateTemplateAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTemplateAliasRequest = exports.UpdateTemplateAliasRequest || (exports.UpdateTemplateAliasRequest = {}));
var UpdateTemplateAliasResponse;
(function (UpdateTemplateAliasResponse) {
    UpdateTemplateAliasResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTemplateAliasResponse = exports.UpdateTemplateAliasResponse || (exports.UpdateTemplateAliasResponse = {}));
var UpdateTemplatePermissionsRequest;
(function (UpdateTemplatePermissionsRequest) {
    UpdateTemplatePermissionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTemplatePermissionsRequest = exports.UpdateTemplatePermissionsRequest || (exports.UpdateTemplatePermissionsRequest = {}));
var UpdateTemplatePermissionsResponse;
(function (UpdateTemplatePermissionsResponse) {
    UpdateTemplatePermissionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTemplatePermissionsResponse = exports.UpdateTemplatePermissionsResponse || (exports.UpdateTemplatePermissionsResponse = {}));
var UpdateThemeRequest;
(function (UpdateThemeRequest) {
    UpdateThemeRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateThemeRequest = exports.UpdateThemeRequest || (exports.UpdateThemeRequest = {}));
var UpdateThemeResponse;
(function (UpdateThemeResponse) {
    UpdateThemeResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateThemeResponse = exports.UpdateThemeResponse || (exports.UpdateThemeResponse = {}));
var UpdateThemeAliasRequest;
(function (UpdateThemeAliasRequest) {
    UpdateThemeAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateThemeAliasRequest = exports.UpdateThemeAliasRequest || (exports.UpdateThemeAliasRequest = {}));
var UpdateThemeAliasResponse;
(function (UpdateThemeAliasResponse) {
    UpdateThemeAliasResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateThemeAliasResponse = exports.UpdateThemeAliasResponse || (exports.UpdateThemeAliasResponse = {}));
var UpdateThemePermissionsRequest;
(function (UpdateThemePermissionsRequest) {
    UpdateThemePermissionsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateThemePermissionsRequest = exports.UpdateThemePermissionsRequest || (exports.UpdateThemePermissionsRequest = {}));
var UpdateThemePermissionsResponse;
(function (UpdateThemePermissionsResponse) {
    UpdateThemePermissionsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateThemePermissionsResponse = exports.UpdateThemePermissionsResponse || (exports.UpdateThemePermissionsResponse = {}));
var UpdateUserRequest;
(function (UpdateUserRequest) {
    UpdateUserRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserRequest = exports.UpdateUserRequest || (exports.UpdateUserRequest = {}));
var UpdateUserResponse;
(function (UpdateUserResponse) {
    UpdateUserResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateUserResponse = exports.UpdateUserResponse || (exports.UpdateUserResponse = {}));
